### DISCUSSION TITLE
Why do ValidationAttributes still have RequiresValidationContext property, when it's not being used?

### DISCUSSION THREAD

**Discussion by kikaragyozov** [dotnet/runtime#43941] [Q&A] [answered]
I couldn't find where that property is ever used in the .NET Core codebase, or ASP.NET Core.

In .NET Framework it's being used only in [here](https://referencesource.microsoft.com/#System.Web.DynamicData/DynamicData/DynamicValidator.cs,162).

**Comment by huoyaoyuan** [comment, marked as answer]
It should be purely for compatibility, because we cannot remove public api, or user code may issue with `TypeLoadException` or `MissingMemberException`.
Having an extra virtual property gives very little harm to the runtime.

**Reply by kikaragyozov** [reply]
Thank you for taking the time to answer my question!
